smarts/core/argoverse_map.py

```diff
@@ -948,11 +964,42 @@ def waypoint_paths(
             road_ids = [road.road_id for road in route.roads]
         if road_ids:
             return self._waypoint_paths_along_route(pose.point, lookahead, road_ids)
+
         closest_lps = self._lanepoints.closest_lanepoints(
             [pose], within_radius=within_radius
         )
-        closest_lane = closest_lps[0].lane
+        closest_lane: RoadMap.Lane = closest_lps[0].lane
+
         waypoint_paths = []
+        if closest_lane.in_junction:
+            junction_lane = closest_lane
+            closest_lanes = [lp.lane for lp in closest_lps]
+            if self._last_junction_lane and self._last_junction_lane in closest_lanes:
```

Comment:
We should be able to assume that `None` isn't in the lanes.
```suggestion
            if self._last_junction_lane in closest_lanes:
```